src/xrpld/ledger/View.h

```diff
@@ -524,18 +567,49 @@ transferXRP(
     STAmount const& amount,
     beast::Journal j);
 
+struct TokenDescriptor
+{
+    std::shared_ptr<SLE const> token;
+    std::shared_ptr<SLE const> issuance;
+};
+
+[[nodiscard]] TER
+requireAuth(ReadView const& view, Issue const& issue, AccountID const& account);
+
 /** Check if the account lacks required authorization.
+ *
  *   Return tecNO_AUTH or tecNO_LINE if it does
  *   and tesSUCCESS otherwise.
  */
 [[nodiscard]] TER
 requireAuth(ReadView const& view, Issue const& issue, AccountID const& account);
+
+/** Check if the account lacks required authorization.
+ *
+ *   Does not check for expired credentials, hence must be followed by
+ *   verifyAuth from doApply
+ */
 [[nodiscard]] TER
 requireAuth(
     ReadView const& view,
     MPTIssue const& mptIssue,
     AccountID const& account);
 
+/** Check if the account lacks required authorization.
+ *
+ *   Called from doApply - it will check for expired (and delete if found any)
+ *   credentials maching DomainID set in MPTIssuance. Must be called if
+ *   requireAuth(...MPTIssue...) check passed in preclaim. Will create MPToken
+ *   if needed, on the basis of non-expired credentals found.
+ */
+[[nodiscard]] TER
+verifyAuth(
```

Comment:
Not sure if it's a good name since it doesn't just verify auth but also creates `MPToken`.

Reply:
renamed to `enforceMPTokenAuthorization` . A bit clumsy but that's best I could come up with :-|